src/bundles/peer-locations.js

```diff
@@ -300,44 +148,65 @@ class PeerLocationResolver {
       ...opts.cache
     })
 
+    this.peersToAddrs = {}
+    this.failedAddrs = []
+
+    this.queue = queue({
+      concurrency: opts.concurrency,
+      autostart: true
+    })
+
     this.geoipLookupPromises = {}
   }
 
-  async lookupWithCaches (peerId, addr, getIpfs) {
+  checkPeer (peerId, addr, getIpfs) {
+    if (this.peersToAddrs[peerId]) {
+      return this.peersToAddrs[peerId]
+    }
+
     const ipv4Tuple = Multiaddr(addr).stringTuples().find(isNonHomeIPv4)
     if (!ipv4Tuple) {
-      throw new Error(`Unable to resolve location for non-IPv4 address ${addr}`)
+      return
+      // throw new Error(`Unable to resolve location for non-IPv4 address ${addr}`)
     }
 
     var ipv4Addr = ipv4Tuple[1]
 
-    // maybe we have it cached by ipv4 address already, check that.
-    let location = await this.geoipCache.get(ipv4Addr)
-    if (location) {
-      return location
+    // ignore lookups that previously failed
+    if (this.failedAddrs.includes(ipv4Addr)) {
+      return
     }
 
     // no ip address cached. are we looking it up already?
     var locPromise = this.geoipLookupPromises[ipv4Addr]
 
     if (!locPromise) {
-      locPromise = new Promise((resolve, reject) => {
-        const ipfs = getIpfs()
-        geoip.lookup(ipfs, ipv4Addr, (err, data) => {
-          // save it in the caches, and remove the promise.
-          this.geoipCache.set(ipv4Addr, data)
-          delete this.geoipLookupPromises[ipv4Addr]
-
-          if (err) return reject(err)
-          resolve(data)
+      this.queue.push(async () => {
+        // maybe we have it cached by ipv4 address already, check that.
+        const location = await this.geoipCache.get(ipv4Addr)
+        if (location) {
+          return location
```

Comment:
What happens to this location? It looks like nothing consumes the result of the queue job, so this ends up just being a little misleading for the reader. I think it's the  peersToAddrs check earlier in the function that would have catches the condition where we'e already looked up this peer.